### PR TITLE
Fix API doc

### PR DESCRIPTION
**Pull request by leezu**
Fixes to docs. See commit descriptions for smaller changes.

The biggest change is to use automodule to generate the API reference of the gluonnlp.data module. Currently classes were listed manually and some were missing (such as the dataset registry utilities).

Please review the generated documentation link below and comment if you see any issues with the result of automodule.

### CODE REVIEW

**Comment by leezu** [comment]
The reason why *Properties* on the Vocab and Embedding API was not rendered correctly is that it is non-standard Sphinx syntax mainly popularized by `numpydoc`. To use it, we should enable the `numpydoc` extension. For now however it causes many many errors related to `toctree contains reference to nonexisting document`, which prevent the documentation to build.

**Comment by mli** [comment]
Job PR-101/10 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-101/10/index.html

**Comment by mli** [comment]
Job PR-101/11 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-101/11/index.html

File: docs/api/embedding.rst [hunk 1]
```diff
@@ -24,10 +24,6 @@ and utility for intrinsic evaluation of text embeddings.
 API Reference
 -------------
 
-.. raw:: html
-
-   <script type="text/javascript" src='../../_static/js/auto_module_index.js'></script>
-
 .. automodule:: gluonnlp.embedding
     :members: register, create, list_sources
```
Comment (szha) [inline docs/api/embedding.rst]: can we use the automodule to list everything inside gluonnlp.embedding too? This way we won't have to worry about listing new classes in API reference

Comment (szha) [reply docs/api/embedding.rst]: same for the API doc for any other module.

Comment (leezu) [reply docs/api/embedding.rst]: I updated the other API references accordingly

**Comment by mli** [comment]
Job PR-101/12 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-101/12/index.html

**Comment by leezu** [comment]
Don't merge yet. The automodule for `gluonnlp.model` did not pick up all docstrings.

**Comment by mli** [comment]
Job PR-101/19 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-101/19/index.html

**Pull request merged**